mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ModClusterSubsystemXMLWriter.java

```diff
@@ -76,10 +77,11 @@ static void writeModClusterConfig(XMLExtendedStreamWriter writer, ModelNode conf
 
     static void writeConfiguration(XMLExtendedStreamWriter writer, ModelNode config) throws XMLStreamException {
         for (AttributeDefinition def : ModClusterConfigResourceDefinition.ATTRIBUTES) {
-            if (def instanceof SimpleAttributeDefinition) {
-                ((SimpleAttributeDefinition) def).marshallAsAttribute(config, true, writer);
+            if (def instanceof StringListAttributeDefinition) {
```

Comment:
Why the conditional marshalling?  Why not just always use: def.getAttributeMarshaller().marshallAsAttribute(def, config, false, writer); ?  Marshalling behavior should be entirely encapsulated within the AttributeMarshaller implementation.  Otherwise, encapsulation is broken.